modules/command-engine/core/src/main/scala/surge/core/commondsl/SurgeCommandBusinessLogic.scala

```diff
@@ -0,0 +1,76 @@
+// Copyright © 2017-2021 UKG Inc. <https://www.ukg.com>
+
+package surge.core.commondsl
+
+import com.typesafe.config.Config
+import io.opentracing.Tracer
+import io.opentracing.noop.NoopTracerFactory
+import surge.core.command.{ AggregateCommandModelCoreTrait, SurgeCommandKafkaConfig }
+import surge.core.{ SurgeAggregateReadFormatting, SurgeWriteFormatting }
+import surge.kafka.KafkaTopic
+import surge.metrics.Metrics
+
+trait SurgeGenericBusinessLogicTrait[AggId, Agg, Command, Rej, Event] {
+
+  protected[surge] val config: Config
+
+  def aggregateName: String
+
+  def stateTopic: KafkaTopic
+  def eventsTopic: KafkaTopic
+
+  def publishStateOnly: Boolean
```

Comment:
Not quite sure how this gets pulled through the different layers, but if possible we should remove this in favor of the new base trait for event processing.